crates/subspace-farmer/src/legacy_multi_plots_farm.rs

```diff
@@ -71,7 +70,7 @@ impl LegacyMultiPlotsFarm {
         let usable_space = get_usable_plot_space(allocated_space);
         let plot_sizes = get_plot_sizes(usable_space, farmer_protocol_info.max_plot_size);
 
-        let first_listen_on: Arc<Mutex<Option<Vec<Multiaddr>>>> = Arc::default();
+        let relay_config = dsn::configure_relay_server(listen_on).await;
```

Comment:
I think relay config should be one of the options provided from the outside, not created internally.

Also the reason I was talking about `node.spawn()` is such that we have direct connection between child and parent and can implement restrictions later by checking IDs of the peers for which to allow relaying and for which to not allow that.

By having just config like this there is no connection back. So I think `node.configure_relay_client()` should actually be something like `node.create_relay_client()`, so that `node` can be provided as an argument where clients need to be created, then `node.configure_relay_client()` can be called to create an instance for particular identity (and returned `node` should store original `node` internally to prevent it from being destroyed and relay being dropped).